kratos/solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h

```diff
@@ -10,465 +10,420 @@
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
     {
         KRATOS_TRY
 
-        mpDofUpdater->UpdateDofs(rDofSet,Dx);
+        mpDofUpdater->UpdateDofs(rDofSet, rDx);
 
         KRATOS_CATCH("")
     }
 
-
-
     /**
-    Performing the update of the solution.
-    */
-    //***************************************************************************
+     * @brief Performing the prediction of the solution.
+     * @param rModelPart The model part of the problem to solve
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
+     */
     void Predict(
-        ModelPart& r_model_part,
+        ModelPart& rModelPart,
         DofsArrayType& rDofSet,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b
-    ) override
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
     {
         KRATOS_TRY
 
         KRATOS_CATCH("")
     }
 
-
     /**
-     * It initializes a non-linear iteration (for the element)
+     * @brief It initializes a non-linear iteration (for the element)
      * @param rModelPart The model of the problem to solve
-     * @param A LHS matrix
-     * @param Dx Incremental update of primary variables
-     * @param b RHS Vector
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
      */
-
     void InitializeNonLinIteration(
         ModelPart& rModelPart,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b
-    ) override
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
     {
         KRATOS_TRY;
 
-        // Initializes the non-linear iteration for all the elements
-        ElementsArrayType& rElements = rModelPart.Elements();
-        ProcessInfo& CurrentProcessInfo = rModelPart.GetProcessInfo();
-
-        const unsigned int NumThreads = OpenMPUtils::GetNumThreads();
-        OpenMPUtils::PartitionVector ElementPartition;
-        OpenMPUtils::DivideInPartitions(rElements.size(), NumThreads, ElementPartition);
-
-        #pragma omp parallel
-        {
-            const unsigned int k = OpenMPUtils::ThisThread();
-
-            typename ElementsArrayType::iterator ElementsBegin = rElements.begin() + ElementPartition[k];
-            typename ElementsArrayType::iterator ElementsEnd   = rElements.begin() + ElementPartition[k + 1];
+        ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
 
-            for (typename ElementsArrayType::iterator itElem = ElementsBegin; itElem != ElementsEnd; itElem++)
-            {
-                itElem->InitializeNonLinearIteration(CurrentProcessInfo);
-            }
+        // Initialize solution step for all of the elements
+        #pragma omp parallel for
```

Comment:
the first loop could have a nowait, but that would be a distinct improveemnt

Reply:
I don't understand, what does the nowait?